frontend/pages/maps/[slug].tsx

```diff
@@ -0,0 +1,104 @@
+import { useState, useEffect } from 'react';
+import { GetStaticPaths, GetStaticProps } from 'next';
+import { fetchData } from 'utils/axios';
+import moment from 'moment';
+import TabView from '@/components/nav-components/tab-nav';
+import campusHashNames from '@/components/data-components/campus-hash.json';
+import { MapPageProps, MapPageData, CampusNameHash } from 'utils/types';
+import config from 'config/config';
+
+const getCampusName = (content: string) => content.slice(0, -1); 
+
+export default function MapPage({ data }: MapPageProps) {
+	type CampusData = {
+		id: string;
+		schedules: {
+			start: string;
+			end: string;
+			subject: string;
+		}[];
+	}[];
+
+	const [campusData, setCampusData] = useState<CampusData>([]);
+
+	useEffect(() => {
+		const campusNameHash: CampusNameHash = campusHashNames
+    async function fetchDataAndUpdateState() {
+			try {
+				const response = await fetchData(`${config.BASE_URL}/GetAllRooms?code=${config.API_KEY}`, 'post', {
+					start: moment(),
+					end: moment().add(9, 'hours'),
+					campus: campusNameHash[getCampusName(data.slug)],
+					intervals: 15,
+				});
+	
+				const formattedCampusData: CampusData = response.data.map(getFormattedCampusData);
+				setCampusData(formattedCampusData);
+			} catch (error) {
+				console.log('fetchDataAndUpdateState ---> AxiosError', error)
+			}
+    }
+
+    function formatCampusData() {
+      const campusData1: CampusData = data.campusData.map(getFormattedCampusData);
```

Comment:
campusData1 is not a good naming.
most of the time for temp values we use sth like _campusData or tempCampusData
but when we put campusData1 as the name of our variable, it seems that we ran out of variable names.